### ISSUE TITLE
Issues using DEBUG var env

### ISSUE THREAD

**Issue by FranciscoMSM** [cypress-io/cypress#5051]
### Current behavior:
When I run `DEBUG=cypress:* cypress run` I got ALL the messages from cypress.
So far so good.
When I try to be more specific ie `DEBUG=cypress:server* cypress run` - I don't get to see any log at all.

### Desired behavior:
According to the docs I should be able to use more specific selectors like the docs ones but they don't seem to work.
![screen](https://user-images.githubusercontent.com/5878425/63875082-9efd2100-c9ba-11e9-85ee-1e811dde20fe.png)

### Steps to reproduce: (app code and test code)
`DEBUG=cypress:server:* cypress run` produces no extra logging

### Versions
Cypress 3.4.0, Debian 9, Electron 61 Headless

**Comment by jennifer-shehane** [comment]
I'm not able to reproduce this issue using Cypress 3.8.1 or 3.4.0, OSX

<img width="768" alt="Screen Shot 2020-01-07 at 2 45 13 PM" src="https://user-images.githubusercontent.com/1271364/71879393-62a5b280-315c-11ea-8847-749e54ce6ce7.png">

Please comment providing more information that may be relevant to why this is happening so that we can investigate.

We use `debug` package for this printing, so maybe it is something involving that. https://github.com/visionmedia/debug

**Comment by jennifer-shehane** [comment]
Reopening with another person mentioning the DEBUG logs not printing for them. https://github.com/cypress-io/cypress/issues/6160

>To reproduce this issue, you can simply start a new cypress project:
>
>`npm i --save cypress`
>`./node_modules/.bin/cypress open`
>
>If you run
>`DEBUG=cypress* node_modules/.bin/cypress run`
>all debugs are printed in the console
>
>if you run
>`DEBUG=cypress:server:browsers:electron node_modules/.bin/cypress run`
>No debug message is printed

>Operating System: Linux Mint

**Comment by jennifer-shehane** [comment]
Closing since still unable to reproduce. Please comment if you're encountering this issue with a description of your system, how you're setting the env vars (maybe even print all env vars) and the output during `cypress run` and we will consider reopening.

**Comment by jennifer-shehane** [comment]
Someone we worked with that had this issue, we narrowed it down to a permissions issue with setting the environment variable in their terminal/shell. You may need to run your terminal in administrative mode or review your permission settings.

**Comment by szy54** [comment]
@jennifer-shehane I am currently facing it and with DEBUG=cypress:server* cypress run it works locally on OSX, however it doesn't work on the ubuntu20.04 EC2 that I usually run the cypress from (as part of CI in jenkins).
I doubt the problem is with permissions to set the env var.
I'll leave couple examples from myself, but I'm not able to dig into cypress or debug code myself to analyze it during $work.
My aim was to enable cypress:server:video logs and you'll laugh at how I achieved it 🤣 
`DEBUG=cypress:*` works
`DEBUG=cypress:cli*` works
`DEBUG=cypress:server*` doesn't work
`DEBUG=cypress:*,cypress:server:video;` prints everything
`DEBUG=cypress:cli*,cypress:server:video` prints only cli*
also I realised that excluding is working
`DEBUG=cypress:*,-cypress:cli*` works as expected
so I ended up with enabling cypress:* and excluding everything else that I don't need. 🤯
`DEBUG=cypress:*,-cypress:launcher*,-cypress:ts*,-cypress:https*,-cypress:proxy*,-cypress:server:u*,-cypress:server:validation*,-cypress:server:r*,-cypress:server:s*,-cypress:server:t*,-cypress:server:p*,-cypress:server:o*,-cypress:server:a*,-cypress:server:b*,-cypress:server:c*,-cypress:server:d*,-cypress:server:e*,-cypress:server:f*,-cypress:server:g*,-cypress:server:h*,-cypress:server:i*,-cypress:server:w*,-cypress:n*,-cypress:c*`
